File: ts/test/smoke.live.mjs
// Manual: node clients/ts/test/smoke.live.mjs  (requires a running server + rs3buddy.json or RS3BUDDY_BASEURL)
import { RS3Buddy } from "../dist/index.js";
const base = process.env.RS3BUDDY_BASEURL;
const buddy = new RS3Buddy(base ? { baseUrl: base } : {});
console.log("baseUrl:", buddy.baseUrl);
const status = await buddy.getStatus();
console.log("status.connected:", status?.connected);
let p = await buddy.getPlayer();
for (let i = 0; i < 8 && !p; i++) p = await buddy.updatePlayer();
console.log("player:", p);

File: python/smoke_live.py
"""Manual: python clients/python/smoke_live.py
Requires a running server + rs3buddy.json, or RS3BUDDY_BASEURL env var.
"""
import os
import sys

sys.path.insert(0, os.path.dirname(__file__))
from rs3buddy import RS3Buddy  # noqa: E402

base = os.environ.get("RS3BUDDY_BASEURL")
buddy = RS3Buddy(base_url=base) if base else RS3Buddy()
print("base_url:", buddy.base_url)
status = buddy.get_status()
print("status.connected:", status.get("connected") if isinstance(status, dict) else status)
p = buddy.get_player()
for _ in range(8):
    if p:
        break
    p = buddy.update_player()
print("player:", p)
buddy.close()
